src/pages/UserSignupRegister.js

```diff
@@ -40,15 +42,20 @@ const Signup = () => {
 
     const handleSubmit = async (e) => {
         e.preventDefault();
-        const { errors, isValid } = validateSignup(formData);
+        const { errors, isValid } = validateFormData(formData);
         setError(errors)
         if (isValid) {
             const { confirmPassword, ...formDataWithoutConfirmPassword } = formData;
-            const res = await dispatch(signupUser(formDataWithoutConfirmPassword));
-            if (res.payload.status === 200) {
+            let res = "";
+            if (user) {
+                res = await dispatch(createUser(formDataWithoutConfirmPassword));
+            } else {
+                res = await dispatch(signupUser(formDataWithoutConfirmPassword));
+            }
```

Comment:
if (user) {
                res = await dispatch(createUser(formDataWithoutConfirmPassword));
            } else {
                res = await dispatch(signupUser(formDataWithoutConfirmPassword));
            }
            
            improve this code @vikash-kanauja